Packs/BloodHoundEnterprise/Integrations/BloodHoundEnterprise/BloodHoundEnterprise_test.py

```diff
@@ -0,0 +1,397 @@
+from freezegun import freeze_time
+import demistomock as demisto
+from unittest.mock import Mock
+import json
+
+
+def load_test_events():
+    with open("test_data/event_list.json") as file:
+        return json.load(file)
+
+
+def create_mock_client(test_events):
+    client = Mock()
+
+    # Mock the search_events method to return slices of the test_events
+    def mock_search_events(limit, from_date, until_date, skip):
+        return test_events[skip: skip + limit]
+
+    client.search_events.side_effect = mock_search_events
+    return client
+
+
+@freeze_time("2024-11-22T13:28:27.698038Z")
+def test_fetch_events_first_time(mocker):
+    """
+    Given:
+        - A mock client with a predefined set of test events loaded from a JSON file.
+        - The current date and time are frozen at "2024-11-22T13:28:27.698038Z".
+        - No previous fetch history from `demisto.getLastRun` (indicating the first time fetching events).
+        - The parameter for `max_events_per_fetch` is set to "7".
+
+    When:
+        - The `fetch_events` function is called to retrieve events with the mock client.
```

Comment:
Make it more clear what the test aims to check. why is it called the "first time"?